### PR TITLE
Implement function pointer calls

### PR DESCRIPTION
**Pull request by zawodskoj**
Varargs are not supported atm - I don't have enough knowledge about them, and they may be target to remake (#285, #196)

## TODO[F]

- [x] TODO sweep

### CODE REVIEW

File: Cesium.CodeGen/Ir/Expressions/IndirectFunctionCallExpression.cs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        if (_calleeType.Parameters?.IsVarArg == true)
+        {
+            // TODO: See https://github.com/ForNeVeR/Cesium/issues/285
+            // Using sparse population of the parameters on the stack. 8 bytes should be enough for anybody.
+            // Also we need perform localloc on empty stack, so we will use local variable to save vararg buffer to temporary variable.
+            if (varArgParametersCount == 0)
+            {
+                scope.AddInstruction(OpCodes.Ldnull);
+            }
+            else
```
Comment (ForNeVeR) [inline Cesium.CodeGen/Ir/Expressions/IndirectFunctionCallExpression.cs:49]: I don't like this copy-paste from the `FunctionCallExpression`. Let's extract it somewhere, or make them lower into each other, or whatever?

**Review by ForNeVeR — CHANGES_REQUESTED** [review]

**Comment by zawodskoj** [comment]
Rebased

File: Cesium.CodeGen/Ir/Expressions/IndirectFunctionCallExpression.cs (hunk 1 above)
Comment (zawodskoj) [reply Cesium.CodeGen/Ir/Expressions/IndirectFunctionCallExpression.cs]: I've extracted that into a base class

**Review by ForNeVeR — APPROVED** [review]
Thanks!

**Pull request merged**